internal-enrichment/urlscan-enrichment/src/main.py

```diff
@@ -1,56 +1,366 @@
+
+                # Orange flag
+                if overall["categories"]:
+                    for categorie in overall["categories"]:
+                        self._create_custom_label(f"urlscan: {categorie}", "#ff801f")
+
+                # Blue flag
+                if overall["brands"]:
+                    for brand in overall["brands"]:
+                        self._create_custom_label(f"urlscan: {brand}", "#5596e3")
+                    # White flag
+                    if "verticals" in overall["brands"]:
+                        for vertical in overall["brands"]["verticals"]:
+                            self._create_custom_label(f"urlscan: {vertical}", "#ffffff")
+
+        return self.all_labels
+
+    def _create_custom_label(self, name_label: str, color_label: str):
+        """
+        This method allows you to create a custom label, using the OpenCTI API.
+
+        :param name_label: A parameter giving the name of the label.
+        :param color_label: A parameter giving the color of the label.
+        """
+
+        new_custom_label = self.helper.api.label.read_or_create_unchecked(
+            value=name_label, color=color_label
+        )
+        if new_custom_label is None:
+            self.helper.connector_logger.error(
+                "[ERROR] The label could not be created. "
+                "If your connector does not have the permission to create labels, "
+                "please create it manually before launching",
+                {"name_label": name_label},
+            )
+        else:
+            self.helper.connector_logger.info(
+                "[INFO] The label has been created.",
+                {"name_label": name_label},
+            )
+            self.all_labels.append(new_custom_label["value"])
+
+    def _generate_stix_bundle(
+        self, data: dict, stix_entity: dict, is_submission: bool
+    ) -> str:
+        """
+        This method create a bundle in Stix2 format.
+
+        :param data: A parameter that contains all the data about the IPv4 that was searched for in URLScan.
+        :param stix_entity: A parameter that contains all the entity information in OpenCTI.
+        :param is_submission: This parameter allows us to identify whether we have a URL or other scopes.
+        :return: str bundle
+        """
+        self.identity = self.converter.generate_urlscan_stix_identity()
+        self.stix_objects.append(self.identity)
+
+        self.helper.connector_logger.info(
+            "[CONNECTOR] The entity has been identified by URLScan "
+            "and generation of the Stix bundle is in progress.",
+            {"entity_type": stix_entity["type"], "entity_value": stix_entity["value"]},
+        )
+
+        if is_submission is True:
+            prepared_file_png = (
+                self.utils.prepare_file_png(data)
+                if self.config.import_screenshot
+                else None
+            )
```

Comment:
When `import_screenshot` configuration is `false`, sending `None` seems to cause an error when enriching an observable

![image](https://github.com/OpenCTI-Platform/connectors/assets/75518128/9ee75c16-2ec7-40f2-a42b-5c9fc46ae25c)

In the worker:
![image](https://github.com/OpenCTI-Platform/connectors/assets/75518128/6847e84f-e250-466b-a6a2-ebf20ba28ea2)